app/models/user.rb

```diff
@@ -36,4 +36,8 @@ def self.calculate_win_ratio(won, finished)
     return 0 unless finished > 0
     (won.to_f / finished * 100).round(2)
   end
+
+  def stale?
+    results.order(created_at: :desc).first.created_at < Time.now.utc - 2.weeks
```

Comment:
I tend not to trust database order as definitively chronological, though 99% of the time it is. This works 100% of the time though :)